### ISSUE TITLE
Build Flask server + Celery workers

### ISSUE THREAD

**Issue by ryuzakinho** [alphacep/vosk-api#265]
Hi,

I have built a Flask server that sends requests to Celery workers to run transcriptions. I do this because the files could be long and I do not want the client to be hanging.

At the moment, each worker has to load its own model. Would you have a recommendation as to how I could improve my approach by using Flask while having only 1 model loaded with concurrency.

**Comment by madkote** [comment]
@ryuzakinho just to clarify (I had similar idea), you store audio locally? since received audio can be long...

**Comment by ryuzakinho** [comment]
@madkote yes, exactly, I upload the file to my server then, trigger a celery task to run the transcription in the background.

**Comment by nshmyrev** [comment]
Celery is not a great design decision here.

**Comment by madkote** [comment]
@nshmyrev agreed. but depends on the use-case - for offline decoding it fits well.

**Comment by madkote** [comment]
@nshmyrev I think the question is, if it is possible to use model with `multiprocessing.shared_memory` - See https://docs.python.org/3/library/multiprocessing.shared_memory.html

**Comment by ryuzakinho** [comment]
Yes, @nshmyrev as @madkote suggested, I am trying to have a small server that is able to perform offline decoding. What setting would you recommend in that case? I am not specifically attached to using Celery but the idea is that I would send a request to Flask endpoint that would trigger a transcription in the background. Parallel transcriptions would leverage one single loaded model.

**Comment by nshmyrev** [comment]
Set a vosk server and send websocket requests from celery to vosk server

**Comment by nshmyrev** [comment]
multiprocessing shared memory is slow due to locking.
